File: c#/JeuDeCarte-C/JeuDeCarte-C/Joueur.cs
namespace CardGame;
//Classe qui fabrique un ou des joueurs
public class Joueur
{
    private static int nbJoueur = 0; // variable pour suivre le prochain numéro unique à un joueur
    private readonly int numJoueur; // numéro unique et fixe du joueur.
    private string nom;
    private string prenom;
    private Paquet mainJoueur; // Main du joueur.

    //Constructeur du joueur
    public Joueur(string nom, string prenom)
    {
        nbJoueur++;
        numJoueur = nbJoueur;
        this.nom = nom;
        this.prenom = prenom;
        mainJoueur = new Paquet();
    }

    public override string ToString()
    {
        return $"Joueur #{numJoueur} : {prenom} {nom}";
    }

    public int GetNumJoueur()
    {
        return numJoueur;
    }

    public string GetNom()
    {
        return nom;
    }

    public void SetNom(string nom)
    {
        this.nom = nom;
    }

    public string GetPrenom()
    {
        return prenom;
    }

    public void SetPrenom(string prenom)
    {
        this.prenom = prenom;
    }

    public Paquet GetMainJoueur()
    {
        return mainJoueur;
    }
}

File: java/JeuDeCarte/src/Joueur.java
public class Joueur {
    private static int nbJoueur = 0; // Incrémenteur de numéro de joueur
    final private int numJoueur; // Numéro séquentiel et unique du joueur
    private String nom;
    private String prenom;
    private Paquet mainJoueur; // Ensemble de cartes du joueur

    public Joueur(String nom, String prenom) {
        nbJoueur++; // Attribut le numéro de joueur suivant
        this.numJoueur = nbJoueur;
        this.nom = nom;
        this.prenom = prenom;
        mainJoueur = new Paquet(); // Création du paquet vide du joueur
    }

    @Override
    public String toString() {
        return "Joueur #" + numJoueur + " : " + prenom + " " + nom ;
    }

    public int getNumJoueur(){
        return numJoueur;
    }
    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public Paquet getMainJoueur() {
        return mainJoueur;
    }

}
